pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go

```diff
@@ -426,6 +433,9 @@ func (sip *streamIngestionProcessor) consumeEvents() (*jobspb.ResolvedSpans, err
 				}
 
 				return sip.flush()
+			case streamingccl.GenerationEvent:
+				log.Info(sip.Ctx, "client disconnected")
+				waitingForCutover = true
```

Comment:
We'd probably also want to wait on ctx.Done() in addition to cutover channel here.

Reply:
I updated the logic here, hope it makes sense now. When a generation event is received, `sip.eventCh` also closes. We wait for `sip.cutoverCh` and `sip.Ctx.Done()`, whichever comes first. If we received value from `sip.cutoverCh`, since the event channel is closed, we flush any remaining events and return. If the context gets cancelled before a cutover signal is sent, the function returns.